### PR TITLE
feat(list editor): Adds support for editing lists (DSP-741)

### PR DESCRIPTION
**Pull request by mdelez**
resolves DSP-741

### CODE REVIEW

**Comment by mdelez** [comment]
@flavens @kilchenmann new edit feature can be found in the app in the list editor of a project and by hovering over a list node :)

**Comment by mdelez** [comment]
oh, also make sure you're using DSP-API 13.1.1!

File: src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        </span>
+        <span class="fill-remaining-space"></span>
+        <span>
+            <button
+                mat-raised-button
+                type="submit"
+                color="primary"
+                [disabled]="saveButtonDisabled"
+                (click)="updateChildNode()">
+                Update
```
Comment (flavens) [inline src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.html:41]: you used `{{ 'appLabels.form.action.cancel' | translate }}` line 30, should use the format `{{ 'appLabels.form.action.update' | translate }}` here as well?! It exists in the `en.json` file

File: src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                this.list = response.body.nodeinfo;
+                this.buildForm(response.body.nodeinfo);
+            },
+            (error: ApiResponseError) => {
+                console.error(error);
+            }
+        );
+    }
+
+    buildForm(list: ListNodeInfo): void {
```
Comment (flavens) [inline src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.ts:59]: would it be possible to add some description for each method, and possibly for tricky code lines?

Comment (mdelez) [reply src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.ts]: added in 2c88e7e

File: src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.html (hunk 1 above)
Comment (mdelez) [reply src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.html]: This was André's code that I moved around. I changed the text to use the translate service for 'update' in 2c88e7e.

File: src/app/project/list/list-item-form/list-item-form.component.ts [hunk 3]
```diff
@@ -86,51 +114,40 @@ export class ListItemFormComponent implements OnInit {
         }
     }
 
-    submitData() {
+    createChildNode() {
```
Comment (flavens) [inline src/app/project/list/list-item-form/list-item-form.component.ts:117]: I will be picky: would it be possible to add in this file as well some method description?

File: src/app/project/list/list-item/list-item.component.ts [hunk 4]
```diff
@@ -54,7 +55,6 @@ export class ListItemComponent implements OnInit {
                 }
             );
         }
-
     }
 
     showChildren(id: string): boolean {
```
Comment (flavens) [inline src/app/project/list/list-item/list-item.component.ts:60]: I will be picky: would it be possible to add in this file as well some method description? even though it is not from you

**Review by flavens — COMMENTED** [review]
Would it be possible to get some more method description? 🙈

File: src/app/project/list/list-item-form/list-item-form.component.ts (hunk 3 above)
Comment (mdelez) [reply src/app/project/list/list-item-form/list-item-form.component.ts:121]: I tried my best as I don't fully understand some of the methods either 😅 added in be62277

File: src/app/project/list/list-item/list-item.component.ts (hunk 4 above)
Comment (mdelez) [reply src/app/project/list/list-item/list-item.component.ts:59]: added in be62277

**Review by flavens — APPROVED** [review]

**Pull request merged**